### PR TITLE
[PM-6484][PM-6535][PM-6486] Revert "[PM-5277] Migrate Sync Service to State Provider (#7680)"

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This reverts commit 78008a9e1e25ce30d641226d3abd37c5b36181e9.

This is to fix PM-6484 where password reprompt was broken in firefox. I honestly do not know what this has to do with reprompt, yet, but bisect pointed to it and I've been unable to reproduce the error since reverting.

The one interesting thing here is that we'd built a bunch of migrations on top of the reverted one, we need to update data past the used version number. I built a builder for noop migrators to do this with some pretty serious warnings around it. This really isn't something we want to do willy nilly.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/state-migrations/migrations/noop-migrator.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { Migrator } from "../migrator";
+
+/**
+ * a no-op migrator that simply allows for updating the version number with no changes to the underlying state.
+ * This is useful when a migration must be removed after other migration have been created.
+ *
+ * @remarks If you need to remove an existing migration, **IT IS REQUIRED** to validate that all following migrations are still valid as they may have been written to build upon the migration that is being removed.
+ *
+ * @remarks **NEVER REMOVE A MIGRATION THAT HAS BEEN RELEASED**
+ */
```
Comment (willmartian) [inline libs/common/src/state-migrations/migrations/noop-migrator.ts:13]: If a migration _was_ released, what would the correct action be?

Comment (willmartian) [reply libs/common/src/state-migrations/migrations/noop-migrator.ts:13]: As we approach quicker releases, there will be less and less time to update an existing-but-not-released migration like this to use this noop migrator, right?

Comment (MGibson1) [reply libs/common/src/state-migrations/migrations/noop-migrator.ts:13]: I think we'd want to fail forward there, or handle by another migrator at the end.

One could argue that that's the more appropriate thing do to here -- rollback this at the end (by making a new migrator that rolls back, not by calling rollback)

**Comment by MGibson1** [comment]
will be tested in `rc`

**Review by willmartian — APPROVED** [review]
Changes LGTM. Thanks for talking through this with me @MGibson1 !

**Review by willmartian — DISMISSED** [review]

**Review by gbubemismith — DISMISSED** [review]

**Comment by MGibson1** [comment]
Note: On this most recent merge I had to move the _other_ migration due to the order of the revert migration being already locked in on the `rc` branch.

**Review by willmartian — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Pull request merged**